### PR TITLE
Add Home Assistant Auto Discovery buttons

### LINKED ISSUE #284: [Feature] Expose a Unlatch button
Hi, it would be cool to have a unlatch / open button exposed too.

### PR DESCRIPTION
**Pull request by iranl**
-Add Home Assistant auto discovery topics for Unlatch, Lock 'n' Go and Lock 'n' Go with Unlatch buttons. Closes #284 and #231

### CODE REVIEW

**Comment by mundschenk-at** [comment]
As laid out in #231, I think this is a terrible idea regarding _unlatch_ (and contrary to HA's architectural decisions). 

Regarding _lock'n'go_ I don't see the use-case? Why would anyone use a virtual button in HA for instead of the physical one on the lock?

**Comment by alexdelprete** [comment]
Personally, I'm in total agreement with this PR.

> As laid out in #231, I think this is a terrible idea regarding _unlatch_ (and contrary to HA's architectural decisions).

You meant #284 probably.

> Regarding _lock'n'go_ I don't see the use-case? Why would anyone use a virtual button in HA for instead of the physical one on the lock?

I rarely touch the lock, I always use the phone: when entering home and when leaving. I already have a button for that. So having an auto-discovered button is easier to configure.

**Comment by mundschenk-at** [comment]
Yeah, #284. 

Well, if people really want to use a virtual button for _lock'n'go_, I have strong aversion against that change. I think it is likely a niche thing, but I could of course be wrong about that. 

I do maintain that any `button` entity that triggers an `unlatch` command should be disabled by default, though, if @technyon decides to include them after all. (By _disabled_ I mean the `enabled_by_default` property should be set to `false` in the discovery payload, not a config option in the web interface.)

**Comment by alexdelprete** [comment]
> if people really want to use a virtual button for _lock'n'go_, I have strong aversion against that change

what does that change cause that you don't like? it's an easier way to configure a button if a user wants to configure it. I really can't understand the rationale of your aversion, let alone the fact that it is a STRONG aversion. It doesn't change anything, it doesn't remove anything, it adds something that is useful to users who want to use the lock'n'go service in a button.

> I do maintain that any `button` entity that triggers an `unlatch` command should be disabled by default

and why should it be disabled by default? because users are stupid? again, you're being paternalistic: I'm a user and I would love to have this button enabled in autodiscovery, why should I have it disabled by default? In order to have a button, users have to still put it on the lovelace card, it's not that automagically it appears and they click it not knowing what it does.

I read zero rational arguments for not merging this PR, only subjective opinion with the prejudice that users are stupid and don't know what they do.

Take into account that users of Nuki Hub are not basic users...it's not a zigbee light...they're using a microcontroller, they update the firmware, they use MQTT, etc. I think they can handle a button better than you think.

**Comment by mundschenk-at** [comment]
> > if people really want to use a virtual button for _lock'n'go_, I have strong aversion against that change
> 
> what does that change cause that you don't like? 

Sorry, typo. I meant to say _**no** strong aversion_.

File: Network.cpp [hunk 1]
```diff
@@ -957,7 +957,8 @@ void Network::publishHASSConfig(char* deviceType, const char* baseTopic, char* n
                          "",
                          "",
                          mqtt_topic_lock_action,
-                         { { "pl_prs", "lockNgo" }});
+                         { { "enabled_by_default", "true" },
```
Comment (mundschenk-at) [inline Network.cpp]: `enabled_by_default` should be set to `false`

Comment (iranl) [reply Network.cpp]: Darn your quick ;)
While I'm not awake apparently

**Pull request merged**